### ISSUE TITLE
AWS credentials using credential_process

### ISSUE THREAD

**Issue by maxbeatty** [apex/up#812]
## Description

If `~/.aws/credentials` uses [credential_process](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html), `up` will fail with a "Cannot find credentials" error.

## Steps to Reproduce

`up.json`

```json
{
  "profile": "developer"
}
```

`~/.aws/credentials`

```
[developer]
credential_process = /opt/bin/awscreds-custom --username helen
```

Try to deploy:
```sh
❯ up

     build: 5 files, 12 MB (545ms)


     Error: Cannot find credentials, visit https://apex.sh/docs/up/credentials/ for help.
```

## Workaround

Setting the environment variable `AWS_SDK_LOAD_CONFIG` will instruct the underlying AWS SDK to use [Shared Config Fields](https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Sessions_from_Shared_Config) (thus executing the custom `credential_process`).

```sh
AWS_SDK_LOAD_CONFIG=1 up
```

---
I'm not sure there's much you could do in your AWS session implementation beyond force-enable Shared Config support. Opening this as mostly as a reference for anyone who comes across the error with this configuration.

**Comment by tj** [comment]
Hmm I wonder if the Go SDK from AWS even supports it, I've never seen that option but I'll try to take a look soon and see if they support it!

**Comment by maxbeatty** [comment]
Yep! It was adding 2018, if you can believe it, and I did check to make sure `up` was using a supported version ([v1.19.1](https://github.com/apex/up/blob/2715f219e15b736745912556a5ecf77a654a85e2/go.mod#L14) for posterity).

https://github.com/aws/aws-sdk-go/commit/275272fc5c7fdea1719f5851925dc5d9df27f89d
https://github.com/aws/aws-sdk-go/releases/tag/v1.16.0

**Comment by irgeek** [comment]
I haven't dug deep enough to create a PR for this, but it looks like a fairly minimal change to enable the shared config by default.

When you're building an AWS SDK session now, you use something like this:

```go
s := session.New(aws.NewConfig().WithRegion(region))
```

But something like this (untested) snippet should load all of the shared config stuff automatically:

```go
sess := session.Must(session.NewSessionWithOptions(session.Options{
    Config: aws.Config{Region: aws.String(region)},
    SharedConfigState: session.SharedConfigEnable,
}))
```

The big behavioural change here is that using `session.NewSessionWithOptions` (with the `session.Must` wrapper) will panic earlier than using `session.New` for some configuration problems. It looks like you're specifically checking for some errors from AWS SDK, so those checks might need updating as well.

Just FYI, the [documentation](https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.38.58/aws/session#New) says `session.New` is deprecated and has been [since 2016](https://github.com/aws/aws-sdk-go/commit/f23263fa57a63e8200f28960672b509c8b0b8232). Using the `session.NewSession*` functions is the preferred way to build sessions.
